src/Umbraco.Web.UI.Client/src/common/filters/truncate.filter.js

```diff
@@ -4,28 +4,52 @@
  * @namespace truncateFilter
  * 
  * param {any} wordwise if true, the string will be cut after last fully displayed word.
- * param {any} max max length of the outputtet string
- * param {any} tail option tail, defaults to: ' ...'
- *
+ * param {any} max where to cut the string.
+ * param {any} tail option tail, defaults to: '…'
+ * 
+ * Legacy version:
+ * parameter noOfChars(wordwise) Where to cut the string.
+ * parameter appendDots(max) If true dots will be appended in the end.
+ * 
  * @description
  * Limits the length of a string, if a cut happens only the string will be appended with three dots to indicate that more is available.
  */
 angular.module("umbraco.filters").filter('truncate', 
     function () {
         return function (value, wordwise, max, tail) {
+            
             if (!value) return '';
+            
+            // Overload-fix to support Forms Legacy Version:
```

Comment:
IMO I think we need more code comments in here to indicate exactly what the heck is going on here. If another developer finds this code they will have no idea what this means and what it is doing. For any hacks like this we need to ensure we have a bunch of code comments in there to make it clear to other developers what is going on. You can include links to these issues too.